### ISSUE TITLE
run mainnet stuck Waiting for Bedrock node to initialize

### ISSUE THREAD

**Issue by m1iktea** [smartcontracts/simple-optimism-node#64]
# Runtime

`.env`
```
###############################################################################
#                                ↓ REQUIRED ↓                                 #
###############################################################################

# Network to run the node on ("mainnet" or "goerli")
NETWORK_NAME=mainnet

# Type of node to run ("full" or "archive"), note that "archive" is 10x bigger
NODE_TYPE=full

# Password for the qBittorrent UI
TORRENT_UI_PASSWORD=adminadmin

###############################################################################
#                            ↓ REQUIRED (LEGACY) ↓                            #
###############################################################################

# Where to sync data from ("l1" or "l2"), see README
SYNC_SOURCE=l1

# Reference L2 node to run healthcheck against
HEALTHCHECK__REFERENCE_RPC_PROVIDER=https://optimism-mainnet.public.blastapi.io

# L1 node to run fault detection against
FAULT_DETECTOR__L1_RPC_PROVIDER=http://192.168.35.216:8550

# Node to get chain data from (if SYNC_SOURCE is "l1" then use L1 node, etc.)
DATA_TRANSPORT_LAYER__RPC_ENDPOINT=http://192.168.35.216:8550

###############################################################################
#                            ↓ REQUIRED (BEDROCK) ↓                           #
###############################################################################

# Where to get the Bedrock database ("download" or "migration"), see README
BEDROCK_SOURCE=download

# L1 node that the op-node (Bedrock) will get chain data from
OP_NODE__RPC_ENDPOINT=http://192.168.35.216:8550

# Type of RPC that op-node is connected to, see README
OP_NODE__RPC_TYPE=basic

###############################################################################
#                                ↓ OPTIONAL ↓                                 #
###############################################################################

# Feel free to customize your image tag if you want, uses "latest" by default
# See here for all avalabe images: https://hub.docker.com/u/ethereumoptimism
IMAGE_TAG__L2GETH=0.5.31
IMAGE_TAG__DTL=0.5.50
IMAGE_TAG__HEALTCHECK=
IMAGE_TAG__FAULT_DETECTOR=0.5.0
IMAGE_TAG__PROMETHEUS=
IMAGE_TAG__GRAFANA=
IMAGE_TAG__INFLUXDB=1.6

# Exposed server ports (must be unique)
# See docker-compose.yml for default values
PORT__L2GETH_HTTP=19991
PORT__L2GETH_WS=19992
PORT__DTL=17878
PORT__HEALTHCHECK_METRICS=17300
PORT__HEALTHCHECK_BEDROCK_METRICS=17302
PORT__FAULT_DETECTOR_METRICS=17301
PORT__PROMETHEUS=
PORT__GRAFANA=
PORT__INFLUXDB=18086
PORT__TORRENT_UI=18080
PORT__TORRENT=16881
PORT__OP_GETH_HTTP=19993
PORT__OP_GETH_WS=19994
PORT__FAULT_DETECTOR_BEDROCK_METRICS=17303
```

`docker-compose.yml`
```yml
---
version: "3.4"
x-logging: &logging
  logging:
    driver: json-file
    options:
      max-size: 10m
      max-file: "3"

services:
  dtl:
    image: ethereumoptimism/data-transport-layer:${IMAGE_TAG__DTL:-latest}
    restart: unless-stopped
    entrypoint: /scripts/dtl-start.sh
    env_file:
      - ./envs/${NETWORK_NAME}/dtl.env
      - .env
    volumes:
      - ./dtl:/db
      - ./scripts/:/scripts/
    ports:
      - ${PORT__DTL:-7878}:7878
    <<: *logging

  l2geth:
    image: ethereumoptimism/l2geth:${IMAGE_TAG__L2GETH:-latest}
    restart: unless-stopped
    stop_grace_period: 3m
    entrypoint: 
      - /bin/sh
      - -c
      - "/scripts/l2geth-init.sh && /scripts/l2geth-start.sh"
    env_file:
      - ./envs/${NETWORK_NAME}/l2geth.env
      - .env
    ports:
      - ${PORT__L2GETH_HTTP:-9991}:8545
      - ${PORT__L2GETH_WS:-9992}:8546
    volumes:
      - ./geth:/geth
      - ./scripts/:/scripts/
    <<: *logging

  healthcheck:
    image: ethereumoptimism/replica-healthcheck:${IMAGE_TAG__HEALTHCHECK:-latest}
    restart: unless-stopped
    env_file:
      - ./envs/${NETWORK_NAME}/healthcheck.env
      - .env
    ports:
      - ${PORT__HEALTHCHECK_METRICS:-7300}:7300
    <<: *logging

  healthcheck-bedrock:
    image: ethereumoptimism/replica-healthcheck:${IMAGE_TAG__HEALTHCHECK:-latest}
    restart: unless-stopped
    env_file:
      - ./envs/${NETWORK_NAME}/healthcheck-bedrock.env
      - .env
    ports:
      - ${PORT__HEALTHCHECK_BEDROCK_METRICS:-7302}:7300
    <<: *logging

  fault-detector:
    image: ethereumoptimism/fault-detector:${IMAGE_TAG__FAULT_DETECTOR:-latest}
    restart: unless-stopped
    env_file:
      - ./envs/${NETWORK_NAME}/fault-detector.env
      - .env
    ports:
      - ${PORT__FAULT_DETECTOR_METRICS:-7301}:7300
    <<: *logging

  fault-detector-bedrock:
    image: ethereumoptimism/fault-detector:${IMAGE_TAG__FAULT_DETECTOR:-latest}
    restart: unless-stopped
    env_file:
      - ./envs/${NETWORK_NAME}/fault-detector-bedrock.env
      - .env
    ports:
      - ${PORT__FAULT_DETECTOR_BEDROCK_METRICS:-7303}:7300
    <<: *logging

  op-geth:
    image: us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:v1.11.2-13ee9ab 
    restart: unless-stopped
    stop_grace_period: 3m
    entrypoint: /scripts/op-geth-start.sh
    env_file:
      - ./envs/${NETWORK_NAME}/op-geth.env
      - .env
    ports:
      - ${PORT__OP_GETH_HTTP:-9993}:8545
      - ${PORT__OP_GETH_WS:-9994}:8546
    volumes:
      - ./scripts/:/scripts
      - ./shared:/shared
      - ./op_geth:/geth
    <<: *logging

  op-node:
    image: us-docker.pkg.dev/oplabs-tools-artifacts/images/op-node:v1.0.0
    restart: unless-stopped
    stop_grace_period: 3m
    entrypoint: /scripts/op-node-start.sh
    env_file:
      - .env
    volumes:
      - ./scripts/:/scripts
      - ./shared:/shared
    <<: *logging

  bedrock-init:
    image: mi1ktea/bedrock-init:foundry-ready
    entrypoint: /scripts/bedrock-init.sh
    env_file:
      - .env
    volumes:
      - ./scripts/:/scripts
      - ./shared:/shared
      - ./op_geth:/geth
      - ./geth:/legacy-geth
      - ./torrent_downloads:/downloads
    <<: *logging

  influxdb:
    image: quay.io/influxdb/influxdb:${IMAGE_TAG__INFLUXDB:-1.6}
    restart: unless-stopped
    env_file:
      - ./envs/${NETWORK_NAME}/influxdb.env
    ports:
      - ${PORT__INFLUXDB:-8086}:8086
    volumes:
      - ./influxdb_data:/var/lib/influxdb

  torrent:
    image: lscr.io/linuxserver/qbittorrent:${IMAGE_TAG__TORRENT:-latest}
    restart: unless-stopped
    environment:
      - PASSWORD=${TORRENT_UI_PASSWORD}
    ports:
      - ${PORT__TORRENT_UI:-8080}:8080
      - ${PORT__TORRENT:-6881}:6881 
      - ${PORT__TORRENT:-6881}:6881/udp
    volumes:
      - ./torrent_config:/config
      - ./torrent_downloads:/downloads
```

# Logs

`op-geth`
```
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
op-geth_1                 | Waiting for Bedrock node to initialize...
```

`op-node`
```
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
op-node_1                 | Waiting for Bedrock node to initialize...
```

`bedrock-init`
```
bedrock-init_1            | ++ echo ''
bedrock-init_1            | + BEDROCK_FLAG=
bedrock-init_1            | + '[' '' == true ']'
bedrock-init_1            | Bedrock flag not set, waiting...
bedrock-init_1            | + echo 'Bedrock flag not set, waiting...'
bedrock-init_1            | + sleep 60
bedrock-init_1            | + true
bedrock-init_1            | ++ config bedrock/mainnet/ready
bedrock-init_1            | +++ cast call 0xcbebc5ba53ff12165239cbb3d310fda2236d6ad2 'config(address,string)(string)' 0x68108902De3A5031197a6eB3b74b3b033e8E8e4d bedrock/mainnet/ready --rpc-url https://goerli.infura.io/v3/84842078b09946638c03157f83405213
bedrock-init_1            | ++ echo ''
bedrock-init_1            | + BEDROCK_FLAG=
bedrock-init_1            | + '[' '' == true ']'
bedrock-init_1            | + echo 'Bedrock flag not set, waiting...'
bedrock-init_1            | + sleep 60
bedrock-init_1            | Bedrock flag not set, waiting...
bedrock-init_1            | + true
bedrock-init_1            | ++ config bedrock/mainnet/ready
bedrock-init_1            | +++ cast call 0xcbebc5ba53ff12165239cbb3d310fda2236d6ad2 'config(address,string)(string)' 0x68108902De3A5031197a6eB3b74b3b033e8E8e4d bedrock/mainnet/ready --rpc-url https://goerli.infura.io/v3/84842078b09946638c03157f83405213
```

torrent_downloads is empty, i think mainnet is not upgraded to Bedrock, Did I make a mistake for running mainnet? 

need help!! Thx

**Comment by Notation** [comment]
same here

**Comment by 0xRook1e** [comment]
same here

**Comment by ZhaoChaoqun** [comment]
same here

**Comment by Chomtana** [comment]
Fixed by #97
